modules/tectonic/resources/manifests/registry/image-registry-rbac.yaml

```diff
@@ -0,0 +1,85 @@
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: cluster-image-registry-operator
+rules:
+- apiGroups:
+  - dockerregistry.operator.openshift.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  - endpoints
+  - events
+  - persistentvolumeclaims
+  - pods
+  - secrets
+  - services
+  verbs:
+  - "*"
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  - deploymentconfigs
+  - deployments
+  - replicasets
+  - statefulsets
+  verbs:
+  - "*"
+- apiGroups:
+  - apps.openshift.io
+  resources:
+  - deploymentconfigs
+  verbs:
+  - "*"
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - "*"
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
```

Comment:
This gives operator `root` on cluster. why not just bind to `cluster-admin` role?

Reply:
one reason i can think of is that this makes it explicit what permissions we actually need/want.

Reply:
> This gives operator root on cluster. why not just bind to cluster-admin role?

We want to explicitly list the permissions that are required. Speaking specifically about this, we need these permissions to create [clusterroles](https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/generate/clusterrole.go#L13). We need to do this dynamically since the number of interfaces registry uses is vary from version to version.